src/components/QuantitySelector/QuantitySelector.jsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from "react";
+import { Button, Container, Controls, Input } from "./QuantitySelector.styled";
+
+export default function QuantitySelector({
+  defaultValue = 1,
+  minValue = 1,
+  maxValue = Infinity,
+  onChange = () => {},
+  dark = false,
+}) {
+  const [quantity, setQuantity] = useState(defaultValue);
+
+  useEffect(() => {
+    onChange(defaultValue);
+  }, [onChange, defaultValue]);
+
+  function changeQuantity(newValue) {
+    newValue = parseInt(newValue);
+    if (isNaN(newValue)) {
+      newValue = quantity;
+    } else if (newValue > maxValue) {
+      newValue = maxValue;
+    } else if (newValue < minValue) {
+      newValue = minValue;
+    }
+    if (newValue !== quantity) {
+      const integerValue = newValue;
```

Comment:
why are we declaring a new variable here ?

Reply:
Sorry I refactored that part along the way and forgot to remove the variable declaration :yum:, fixed on 6116fad